src/Fixer/Whitespace/NoExtraConsecutiveBlankLinesFixer.php

```diff
@@ -91,7 +91,13 @@ public function configure(array $configuration = null)
                     $this->tokenKindCallbackMap[CT_USE_TRAIT] = 'removeBetweenUse';
                     break;
                 case 'curly_brace_open':
-                    $this->tokenEqualsMap['{'] = 'fixAfterToken';
+                    $this->tokenEqualsMap['{'] = 'fixAfterTokenIfMultiLineBody'; // i.e. not: CT_ARRAY_INDEX_CURLY_BRACE_OPEN
+                    break;
+                case 'square_brace_open':
+                    $this->tokenKindCallbackMap[CT_ARRAY_SQUARE_BRACE_OPEN] = 'fixAfterTokenIfMultiLineBody'; // typeless '[' tokens should not be fixed (to rare)
```

Comment:
for me it's a must have if we want to fix blank line after opening array by `[` we need to fix it after opening array be `array(`.
before closing would be nice as well, and actually could be single option together with after opening.